File: check-collections.py
import sqlite3

db = sqlite3.connect('electron-app/app-data/database/app.db')
cursor = db.cursor()

# Get collections
cursor.execute('''
    SELECT fc.collection_id, fc.name,
           COUNT(DISTINCT f.image_id) as images,
           COUNT(f.face_id) as faces
    FROM face_collections fc
    LEFT JOIN face_collection_members fcm ON fc.collection_id = fcm.collection_id
    LEFT JOIN faces f ON fcm.face_id = f.face_id
    GROUP BY fc.collection_id
    ORDER BY faces DESC
''')

collections = cursor.fetchall()

print(f'=== TOTAL COLLECTIONS: {len(collections)} ===\n')

for i, c in enumerate(collections):
    print(f'{i+1}. {c[1]} ({c[0][:8]}...)')
    print(f'   - {c[2]} images with this person')
    print(f'   - {c[3]} faces total\n')

# Get processing status
cursor.execute('SELECT COUNT(*) FROM images WHERE processing_status = "completed"')
completed = cursor.fetchone()[0]
print(f'Processed: {completed}/7 images')

db.close()

File: electron-app/check-collections.js
const Database = require('better-sqlite3');
const path = require('path');

const dbPath = path.join(__dirname, 'app-data', 'database.db');
const db = new Database(dbPath);

// Count total collections
const countResult = db.prepare('SELECT COUNT(*) as count FROM face_collections').get();
console.log('Total collections in local DB:', countResult.count);

// Get most recent collections
const recentCollections = db.prepare(`
  SELECT collection_id, name, created_at
  FROM face_collections
  ORDER BY created_at DESC
  LIMIT 5
`).all();

console.log('\nMost recent collections:');
recentCollections.forEach((col, idx) => {
  console.log(`${idx + 1}. ${col.name} (${col.collection_id.substring(0, 12)}...) - Created: ${col.created_at}`);
});

// Check if there are any collections created today
const today = new Date().toISOString().split('T')[0];
const todayCollections = db.prepare(`
  SELECT COUNT(*) as count
  FROM face_collections
  WHERE created_at LIKE ?
`).get(`${today}%`);

console.log(`\nCollections created today (${today}):`, todayCollections.count);

// Check GCS sync status
const unsyncedCount = db.prepare(`
  SELECT COUNT(*) as count
  FROM face_collections
  WHERE gcs_synced_at IS NULL OR gcs_synced_at = ''
`).get();

console.log('Collections not synced to GCS:', unsyncedCount.count);

db.close();
